2009-03-01  Aurelien Jarno  <aurelien@aurel32.net>

	* timezone/Makefile: Call cross-zic directly instead of using
	the target ld.so while generating testdata.

Index: timezone/Makefile
===================================================================
--- a/timezone/Makefile	(revision 8009)
+++ b/timezone/Makefile	(working copy)
@@ -202,10 +202,17 @@
 # Don't add leapseconds here since test-tz made checks that work only without
 # leapseconds.
 testdata = $(objpfx)testdata
+ifeq (no,$(cross-compiling))
 define build-testdata
 GCONV_PATH=${common-objpfx}iconvdata LANGUAGE=C LC_ALL=C \
   $(built-program-cmd) -d $(testdata) -y ./yearistype $<
 endef
+else
+define build-testdata
+LANGUAGE=C LC_ALL=C \
+  $(objpfx)cross-zic -d $(testdata) -y ./yearistype $<
+endef
+endif
 
 $(objpfx)test-tz.out: $(addprefix $(testdata)/, America/New_York Etc/UTC UTC)
 $(objpfx)tst-timezone.out: $(addprefix $(testdata)/, \
